docs/development/release_guide.rst

```diff
@@ -105,3 +109,31 @@ Post-release
 * Send release announcement to mailing list
 
 * Update the release guide to reflect any changes
+
+Compatibility with Prior Versions of Python, NumPy, and Astropy
+===============================================================
+
+PlasmaPy releases will generally abide by the following standards,
+which are adapted from `NumPy Enhancement Proposal 29
+<https://numpy.org/neps/nep-0029-deprecation_policy.html>`_ for the
+support of old versions of Python, NumPy, and Astropy.
+
+* PlasmaPy should support at least the minor versions of Python
+  initially released 42 months prior to a planned project release date.
+* PlasmaPy should support at least the 2 latest minor versions of
+  Python.
+* PlasmaPy should support minor versions of NumPy initially released
+  in the 24 months prior to a planned project release date or the
+  oldest version that supports the minimum Python version (whichever is
+  higher).
+* PlasmaPy should support at least the 3 latest minor versions of
+  NumPy and Astropy.
+
+Upstream package requirements should only change during major or minor
+releases, and never during a patch release.  All supported minor
+versions of Python should be in the test matrix and have binary
+artifacts built for releases.
```

Comment:
As of now, we don't actually have binary artifacts, since we got rid of Cython, we're running on conda in so-called "noarch" mode as a pure Python dependency.